### PR TITLE
Handle multiple spappresourcedirs in reports and resource end points

### PR DESCRIPTION
**Pull request by realVinayak**
At least at two places backend assumes that there is just one spappresourcedir. Hence, it uses django get() which raises exception on more than one dir returned. 
Since the app resources use a filter. Fixed by using filter() and then taking arbitrary appresourcedir. Since appresourcedirs are essentially identical (that's why you got them more in the first place), arbitrary selection would work.

Also fixes a bug for statspage that @carlosmbe reported

### CODE REVIEW

**Comment by grantfitzsimmons** [comment]
Works on New Brunswick's database when creating a report or label.

**Review by grantfitzsimmons — APPROVED** [review]
Solves the issues on both databases we experienced issues within my testing. Able to create reports and labels, use stats page, personal prefs, created app resources, forms, etc. without issue.

Need another review from @specify/ux-testing and from a @specify/dev-testing member

File: specifyweb/context/resources.py
Comment (realVinayak) [reply specifyweb/context/resources.py]: Actually this is incorrect. The filter and create are actually two different dictionaries but django handled it correctly. Using dictionaries (in the commit I referenced before) broke it. Fixed it again by separating filter and create

**Review by maxxxxxdlp — APPROVED** [review]

**Review by melton-jason — APPROVED** [review]
Code looks good!
I am not the biggest fan of using `len(directories_matches: QuerySet)` to get the count. `QuerySet.count()` is more performant (https://docs.djangoproject.com/en/4.2/ref/models/querysets/#count), 
However, doing the `len()` operation caches the results of the QuerySet. 
Normally, accessing an element by indexing a QuerySet (i.e., `directories_matched[0]`) would hit the database again to fetch the object. 
If you're concerned about eeking out every ounce of performance, I would look and see if accessing an element by index from a QuerySet tries to access the QuerySet cache, or if it just directly goes to the database. 

Of course, this would have a very little impact on performance anyways, so I will leave it up to you.

**Pull request merged**
